File: genefunk/src/main/java/de/hipp/pnp/genefunk/GeneFunkClassRepository.java
package de.hipp.pnp.genefunk;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
interface GeneFunkClassRepository extends JpaRepository<GeneFunkClass, Integer> {
	Optional<GeneFunkClass> findByName(String name);
}

File: genefunk/src/main/java/de/hipp/pnp/genefunk/GeneFunkClassRepository.kt
package de.hipp.pnp.genefunk

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface GeneFunkClassRepository : JpaRepository<GeneFunkClassEntity?, Int?> {
    fun findByName(name: String?): Optional<GeneFunkClassEntity?>?
}
